text/1001-websocket-protocol.rst

```diff
@@ -0,0 +1,139 @@
+..
+    Status: Draft
+    Type: Feature
+    Created: 2020-02-19
+    RFC PR: `edgedb/rfcs#4 <https://github.com/edgedb/rfcs/pull/4>`_
+
+======================================
+RFC 0001: The WebSocket-based Protocol
+======================================
+
+This RFC describes the design and implementation of the alternate
+protocol to the EdgeDB based on WebSockets for framing and the initial
+handshake.
+
+
+Motivation
+==========
+
+The primary motivation is to allow various browser-based exploration
+and editing tools for the database.
+
+The secondary motivation is to allow browser-based access to the
+application data. Essentially allowing serverless implementation for
+some applications or parts thereof. The reason this is a secondary
+motivation is that it requires full featured access controls (ACLs)
+which we don't have yet.
+
+Two important side effects of implementing this RFC:
+* it might also help implementing the protocol in some languages which
+  we don't have language bindings yet
```

Comment:
It might help marginally, but I think that the hardest part of the binding implementation is not in how messages are framed, but actually handling the message payloads according to spec, which includes parsing type descriptors, decoding/encoding of data, handling authentication etc.

Reply:
Sure, it helps only marginally, that's why it's only listed under "side effects" title not as the primary motivation.